api/src/main/java/org/openmrs/module/appointmentscheduling/api/db/hibernate/HibernateAppointmentDAO.java

```diff
@@ -237,6 +238,40 @@ public Integer getCountOfAppointmentsInTimeSlotByStatus(TimeSlot timeSlot,
 				.intValue();
 	}
 
+	@Override
+	public List<Appointment> getDefaultersList(int minDays, int maxDays, Provider provider, AppointmentType appointmentType, VisitType visitType) throws DAOException {
+
+		String stringQuery = "SELECT  appointment FROM Appointment AS appointment WHERE appointment.status = :status AND appointment.voided = 0" +
+
+				"   AND datediff(curdate(), date_format(appointment.timeSlot.endDate, '%Y-%m-%d')) >= :mindays" +
+				"   AND datediff(curdate(), date_format(appointment.timeSlot.endDate, '%Y-%m-%d')) <= :maxdays";
+
```

Comment:
what if mindays and/or maxdays are null?

Reply:
i did the checks on the resource